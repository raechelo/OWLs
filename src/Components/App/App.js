import React, { Component } from 'react';
import data from '../../data'
import Header from '../Header/Header';
import QuizContainer from '../QuizContainer/QuizContainer'

class App extends Component {
  constructor() {
    super()
    this.state = {
      questions: [],
      house: ''
    }
  }

  componentDidMount = () => {
    this.setState( {questions: data} )
  }

  handleClick = (e) => {
    this.setState( {house: e.target.textContent} )
  }


  render() {
    let defaultPage = 
    <div className="App">
      <section className="btn-container">
        <h2>Choose Your House</h2>
        <button onClick={this.handleClick} className="gryf btn">gryffindor</button>
        <button onClick={this.handleClick} className="huf btn">hufflepuff</button>
        <button onClick={this.handleClick} className="rav btn">ravenclaw</button>
        <button onClick={this.handleClick} className="sly btn">slytherin</button>
      </section>
    </div>
  
    return (
      <div className="style">
      <Header />
      {this.state.house ? <QuizContainer studySet={this.state.questions} /> : defaultPage}
      </div>
    )
  }
}

export default App;