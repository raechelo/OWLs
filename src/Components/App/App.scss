 body {
  background-image: url('./assets/#{$nth}.jpg');
  background-size: cover;
  background-repeat: no-repeat;
}



.rav {
  background-color: $rav-blue;
  border: 3px solid $rav-silver;
  color: $rav-silver;
  font-family: $bodyfont;
  font-size: 1.5rem;
}

.gryf {
  background-color: $gryf-red;
  border: 3px solid $gryf-gold;
  color: $gryf-gold;
  font-family: $bodyfont;
  font-size: 1.5rem;
}

.huf {
  background-color: $huf-yellow;
  border: 3px solid $huf-black;
  color: $huf-black;
  font-family: $bodyfont;
  font-size: 1.5rem;
}

.sly {
  background-color: $sly-green;
  border: 3px solid $sly-silver;
  color: $sly-silver;
  font-family: $bodyfont;
  font-size: 1.5rem;
}

.btn-container {
  background-color: rgba(0, 0, 0, .7);
  display: grid;
  margin: 1rem auto;
  text-align: center;
  width: 50%;

  h2 {
    color: rgba(128, 124, 128, 1);
    font-family: $bodyfont;
  }
}

.btn {
  margin: 1rem;
  padding: .5rem;
  text-transform: uppercase;
}