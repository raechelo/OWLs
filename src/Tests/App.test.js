import React from 'react';
import ReactDOM from 'react-dom';
import App from '../Components/App/App';
import { shallow } from 'enzyme';



describe('App', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <App />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should have a proper default state', () => {
    expect(wrapper.state()).toEqual( {questions: [], incorrect: [], house: ''} )
  });

  it('should update the house on user choice', () => {
    expect(wrapper.state('house')).toEqual('');
    wrapper.instance().handleClick( { target: { textContent: 'ravenclaw' } } );
    expect(wrapper.state('house')).toEqual('ravenclaw');
  });
})
