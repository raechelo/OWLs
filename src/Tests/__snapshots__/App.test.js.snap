// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all the data passed in 1`] = `
<div
  className="style"
>
  <Header />
  <div
    className="App"
  >
    <section
      className="btn-container"
    >
      <h2>
        Choose Your House
      </h2>
      <button
        className="gryf btn"
        onClick={[Function]}
      >
        gryffindor
      </button>
      <button
        className="huf btn"
        onClick={[Function]}
      >
        hufflepuff
      </button>
      <button
        className="rav btn"
        onClick={[Function]}
      >
        ravenclaw
      </button>
      <button
        className="sly btn"
        onClick={[Function]}
      >
        slytherin
      </button>
    </section>
  </div>
</div>
`;
